js/ui/tree_view.js

```diff
@@ -1547,9 +1547,9 @@ var TreeView = HierarchicalCollectionWidget.inherit({
             return;
         }
 
-        var $target = $(e.target).closest("." + NODE_CLASS);
+        var $target = $(e.target).closest("." + NODE_CLASS + ", ." + SELECT_ALL_ITEM_CLASS);
 
-        if(!$target.hasClass(NODE_CLASS)) {
+        if(!$target.length) {
             return;
         }
 
```

Comment:
Please check if we can return focus to the first node using keyboard when selectAll item is focused